jaxrs-clients/src/main/java/com/palantir/remoting3/jaxrs/AbstractFeignJaxRsClientBuilder.java

```diff
@@ -62,19 +62,26 @@
 abstract class AbstractFeignJaxRsClientBuilder {
 
     private final ClientConfiguration config;
+    /**
+     * The primary URI used to bootstrap the Feign client; this is the URI used by Feign to create an OkHttp call. Note
+     * that the {@link MultiServerRetryInterceptor} replaces this URI with a random URI from the client configuration
+     * before making the request.
+     */
+    private final String primaryUri;
 
     AbstractFeignJaxRsClientBuilder(ClientConfiguration config) {
-        this.config = config;
         Preconditions.checkArgument(config.maxNumRetries() == 0,
                 "Connection-level retries are not supported by %s", JaxRsClient.class.getSimpleName());
+        Preconditions.checkArgument(!config.uris().isEmpty(), "Must provide at least one service URI");
+        this.config = config;
+        this.primaryUri = config.uris().get(new Random().nextInt(config.uris().size()));
```

Comment:
if this isn't going to be used, then we should just pick the first one, just cos it's irrelevant (the reader would assume this is in some way relevant).